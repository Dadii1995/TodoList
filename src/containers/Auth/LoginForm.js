import React from 'react'
import withAuth from '../../hocs/withAuth'
import { Button } from 'reactstrap'
import { Redirect } from 'react-router-dom'

const LoginForm = props => {
  return props.isAuth ? (
    <Redirect to="/" />
  ) : (
    <Button
      data-cy="login-button"
      onClick={() => {
        props.logIn()
      }}
    >
      Log in
    </Button>
  )
}

export default withAuth(LoginForm)
