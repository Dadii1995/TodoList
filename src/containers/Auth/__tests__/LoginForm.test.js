import React from 'react'
import renderer from 'react-test-renderer'
import LoginForm from '../LoginForm'

const getWrapper = () => {
  const wrapper = renderer.create(<LoginForm />)
  const root = wrapper.root
  return { root, wrapper }
}
describe('<LoginForm/>', () => {
  test('should render login form', () => {
    const { wrapper } = getWrapper()

    expect(wrapper).toMatchSnapshot()
  })
})
