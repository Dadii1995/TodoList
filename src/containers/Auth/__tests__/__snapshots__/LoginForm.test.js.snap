// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm/> should render login form 1`] = `
<button
  aria-label={null}
  className="btn btn-secondary"
  data-cy="login-button"
  onClick={[Function]}
  type="button"
>
  Log in
</button>
`;
